File: lingopulse-mobil/mobile_app/src/types/review.ts
export interface ReviewSubmit {
  word_id: string;
  knew_it: boolean;
}

export interface ReviewResponse {
  id: string;
  word_id: string;
  knew_it: boolean;
  previous_level: number;
  new_level: number;
  previous_interval_days: number;
  new_interval_days: number;
  reviewed_at: string;
}

File: backend/app/domain/schemas/review.py
from datetime import datetime

from pydantic import BaseModel


class ReviewSubmit(BaseModel):
    word_id: str
    knew_it: bool


class ReviewResponse(BaseModel):
    id: str
    word_id: str
    knew_it: bool
    previous_level: int
    new_level: int
    previous_interval_days: int
    new_interval_days: int
    reviewed_at: datetime
